NullPointerException in TableDump.formatLine

We are using slick 3.0.1 & slick-pg 0.9.2. Sometimes we see this exception in our logs:

```
2015-10-09 19:16:34.627807500 Caused by: java.lang.NullPointerException: null
2015-10-09 19:16:34.627807500 	at slick.util.TableDump$$anonfun$formatLine$1.apply(TableDump.scala:15) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627808500 	at slick.util.TableDump$$anonfun$formatLine$1.apply(TableDump.scala:14) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627813500 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627814500 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627815500 	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627818500 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627819500 	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627820500 	at scala.collection.AbstractTraversable.map(Traversable.scala:104) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627823500 	at slick.util.TableDump.formatLine(TableDump.scala:14) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627824500 	at slick.util.TableDump$$anonfun$6.apply(TableDump.scala:20) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627828500 	at slick.util.TableDump$$anonfun$6.apply(TableDump.scala:20) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627829500 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627830500 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627833500 	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627834500 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627835500 	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627838500 	at scala.collection.AbstractTraversable.map(Traversable.scala:104) ~[org.scala-lang.scala-library-2.11.7.jar:na]
2015-10-09 19:16:34.627839500 	at slick.util.TableDump.apply(TableDump.scala:20) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627840500 	at slick.jdbc.StatementInvoker$$anon$2.close(StatementInvoker.scala:54) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627844500 	at slick.jdbc.PositionedResultIterator.close(PositionedResult.scala:188) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
2015-10-09 19:16:34.627845500 	at slick.jdbc.Invoker$class.foreach(Invoker.scala:70) ~[com.typesafe.slick.slick_2.11-3.0.1.jar:na]
```

I did not found exact cause of this problem, possiblty it is related to reading nullable array field.

TableDump is a debug function, I think it should not crash when `value.toString` returns null.

--- a/slick/src/main/scala/slick/util/TableDump.scala
+++ b/slick/src/main/scala/slick/util/TableDump.scala
@@ -12,7 +12,8 @@ class TableDump(maxColumnWidth: Int = 20) {
   protected[this] val spaces = Iterator.fill(maxColumnWidth+2)(' ').mkString
 
   protected[this] def formatLine(line: IndexedSeq[Any]): IndexedSeq[String] = line.map { v =>
-    (if(v == null) "NULL" else v.toString).replace("\r", "\\r").replace("\n", "\\n").replace("\t", "\\t")
+    val s = if(v == null) "NULL" else v.toString
+    if(s == null) "NULL" else s.replace("\r", "\\r").replace("\n", "\\n").replace("\t", "\\t")
   }
 
   def apply(headers: IndexedSeq[IndexedSeq[String]], data: IndexedSeq[IndexedSeq[Any]]): IndexedSeq[String] = {

